### PR TITLE
fixed csv export fails

### PR DESCRIPTION
**Pull request by Ohior**
Now The csv file will save whether or not scoped storage is ignored or not

### CODE REVIEW

File: app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt [hunk 1]
```diff
@@ -130,18 +141,32 @@ class MainActivity : AbstractBaseActivity(), SwipeRefreshLayout.OnRefreshListene
...
-                val filename = "prepaid-balance-${System.currentTimeMillis()}.csv"
                 writeToFileInDownloads(content, filename)
                 showSnackbar(getString(R.string.export_saved_file, filename))
                 return@launch
-            } catch (e: Exception) {
-                Log.e(TAG, "Error saving file", e)
+            } catch (e: FileNotFoundException) {
+                if (hasPermissions(READ_EXTERNAL_STORAGE) && hasPermissions(WRITE_EXTERNAL_STORAGE)){
+                    exportAsCsv()
+                }else{
```
Comment (mueller-ma) [inline app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt:166]:
```suggestion
                } else {
```

File: app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt [hunk 2]
```diff
@@ -130,18 +141,32 @@ class MainActivity : AbstractBaseActivity(), SwipeRefreshLayout.OnRefreshListene
...
+//            try {
+//                val filename = "prepaid-balance-${System.currentTimeMillis()}.csv"
+//                writeToFileInDownloads(content, filename)
+//                showSnackbar(getString(R.string.export_saved_file, filename))
+//                return@launch
+//            } catch (e: Exception) {
+//                Log.e(TAG, "Error saving file", e)
+//            }
+//
+//            showSnackbar(R.string.export_error_saving_file)
```
Comment (mueller-ma) [inline app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt:155]: Please remove this.

File: app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt [hunk 3]
```diff
@@ -57,6 +58,16 @@ class MainActivity : AbstractBaseActivity(), SwipeRefreshLayout.OnRefreshListene
         }
     }
 
+    private val requestStoragePermission = registerForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) { permission ->
+        val isRead = permission[READ_EXTERNAL_STORAGE]?:false
+        val isWrite = permission[WRITE_EXTERNAL_STORAGE]?:false
```
Comment (mueller-ma) [inline app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt:63]:
```suggestion
        val canRead = permission[READ_EXTERNAL_STORAGE] ?: false
        val canWrite = permission[WRITE_EXTERNAL_STORAGE] ?: false
```

File: app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt [hunk 4]
```diff
@@ -57,6 +58,16 @@ class MainActivity : AbstractBaseActivity(), SwipeRefreshLayout.OnRefreshListene
...
+        if (isRead && isWrite){
+            exportAsCsv()
+        }else{
+            showSnackbar("Permission Denied")
```
Comment (mueller-ma) [inline app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt:67]:
```suggestion
            showSnackbar(R.string.export_error_saving_file)
```

**Review by mueller-ma — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt (hunk 2 above)
Comment (Ohior) [reply app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt:155]: I was preventing myself from removing any piece of your code. That was why I commented it and not delete it.

**Pull request closed without merging**